### PR TITLE
remove layers.tensor.argmin/argmax/assign/cast/concat/sums

### PR DESCRIPTION
**Pull request by mhy-666**
### PR types

Others
### PR changes

APIs
### Describe

To clean the code in paddle.fluid, the useless codes should be removed and code still be used should move to a new position.

For function paddle.fluid.layers.tensor.argmin/argmax/assign/cast/concat/sums, since it is not used in version 2.x, should be removed.

For unittest code where this function is used, if it is aimed to test the old API, it can be removed directly. Otherwise, it should be replaced with paddle.argmin/argmax/assign/cast/concat/sums. And please note names of parameter are slightly different.

### CODE REVIEW

File: python/paddle/incubate/autograd/primitives.py [hunk 1]
```diff
@@ -11,8 +11,6 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from paddle.fluid.layers.tensor import assign  # noqa: F401
-from paddle.fluid.layers.tensor import cast  # noqa: F401
```
Comment (zoooo0820) [inline python/paddle/incubate/autograd/primitives.py:15]: 这里的内容和一般API不太一样，请将`assign`和`cast`换成从2.0引用，并且不要在`others`中删除对应项

File: python/paddle/nn/clip.py [hunk 2]
```diff
@@ -766,7 +766,7 @@ def _static_clip(self, params_grads):
                     else:
                         global_norm_var.append(global_norm_var_fp16)
                 if len(sum_square_list_fp32) > 0:
-                    global_norm_var_fp32 = layers.sums(sum_square_list_fp32)
+                    global_norm_var_fp32 = paddle.sum(sum_square_list_fp32)
```
Comment (zoooo0820) [inline python/paddle/nn/clip.py]: `fluid.layers.sums` -> `paddle.add_n`

**Review by zoooo0820 — COMMENTED** [review]
`python/paddle/fluid/framework.py`中`clear_gradient`有文档内容涉及`fluid.sums`，需要一并修改，否则示例代码会报错

File: python/paddle/nn/clip.py (hunk 2 above)
Comment (mhy-666) [reply python/paddle/nn/clip.py]: ok

File: python/paddle/tensor/__init__.py [hunk 3]
```diff
@@ -291,6 +292,7 @@
 
 # this list used in math_op_patch.py for _binary_creator_
 tensor_method_func = [  # noqa
+    'assign',
```
Comment (zoooo0820) [inline python/paddle/tensor/__init__.py]: 目前没有`Tensor.assign`的调用方式，本次迁移中保持前后一致就行，不需要在这个文件中做改动

Comment (mhy-666) [reply python/paddle/tensor/__init__.py]: done

**Review by zoooo0820 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by dingjiaweiww — APPROVED** [review]

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Review by YuanRisheng — APPROVED** [review]

**Review by zhupengyang — APPROVED** [review]
LGTM

**Pull request merged**